File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/CallbackQueryFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface CallbackQueryFilter {

    MessageFilter message() default @MessageFilter(status = AnnotationState.OFF);

    StringFilter inlineMessageId() default @StringFilter(status = AnnotationState.OFF);

    StringFilter chatInstance() default @StringFilter(status = AnnotationState.OFF);

    StringFilter data() default @StringFilter(status = AnnotationState.OFF);

    StringFilter gameShortName() default @StringFilter(status = AnnotationState.OFF);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/annotations/filter/composite/CallbackQueryFilter.kt
package io.github.oleksivio.tl.kbot.core.annotations.filter.composite

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.StringFilter
import io.github.oleksivio.tl.kbot.core.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class CallbackQueryFilter(
    val from: UserFilter = UserFilter(
        status = AnnotationState.OFF
    ),
    val message: MessageFilter = MessageFilter(
        status = AnnotationState.OFF
    ),
    val validator: Array<String> = [],
    val inlineMessageId: StringFilter = StringFilter(
        status = AnnotationState.OFF
    ),
    val chatInstance: StringFilter = StringFilter(
        status = AnnotationState.OFF
    ),
    val data: StringFilter = StringFilter(
        status = AnnotationState.OFF
    ),
    val gameShortName: StringFilter = StringFilter(
        status = AnnotationState.OFF
    )
)
